etna/analysis/feature_relevance/relevance_table.py

```diff
@@ -39,11 +40,31 @@ def get_statistics_relevance_table(df: pd.DataFrame, df_exog: pd.DataFrame) -> p
     regressors = sorted(df_exog.columns.get_level_values("feature").unique())
     segments = sorted(df.columns.get_level_values("segment").unique())
     result = np.empty((len(segments), len(regressors)))
+    none_warning_raised = False
+    category_warning_raised = False
     for k, seg in enumerate(segments):
         first_valid_idx = df.loc[:, seg].first_valid_index()
         df_now = df.loc[first_valid_idx:, seg]["target"]
-        df_exog_now = df_exog.loc[first_valid_idx:, seg]
-        relevance = calculate_relevance_table(df_exog_now[: len(df_now)], df_now)[["feature", "p_value"]].values
+        df_exog_now = df_exog.loc[first_valid_idx:, seg][: len(df_now)]
+        cat_cols = df_exog_now.dtypes[df_exog_now.dtypes == "category"].index
+        for cat_col in cat_cols:
+            try:
+                df_exog_now[cat_col] = df_exog_now[cat_col].astype(float)
+            except ValueError:
+                raise ValueError(f"{cat_col} column cannot be cast to float type! Please, use encoders.")
+        if len(cat_cols) > 0 and not category_warning_raised:
+            category_warning_raised = True
+            warnings.warn(
+                "Exogenous data contains columns with category type! It will be converted to float. If this is not desired behavior, use encoders."
+            )
+        df_exog_now["target"] = df_now
+        df_exog_now = df_exog_now.dropna()
```

Comment:
Let's drop nans the same way as [here](https://github.com/tinkoff-ai/etna/blob/b65bfb72203ad0e060ad93181edde0065dfe5dd1/etna/analysis/feature_relevance/relevance_table.py#L75) to make this step similar in both methods